bin/git-ignore

```diff
@@ -35,6 +35,16 @@ function add_local {
   add_patterns .gitignore "$@"
 }
 
+function show_private {
+  cd "$(git root)"
+  show_contents Private .git/info/exclude
+}
+
+function add_private {
+  cd "$(git root)"
+  add_patterns .git/info/exclude "$@"
```

Comment:
@flatcap 
I think you need to create `.git/info` first if the directory doesn't exist.

Reply:
Hmm... I never would have guessed that the `info` dir was optional!
I've added a `mkdir -p` to be sure.

Reply:
@flatcap 
Better to check if the directory is existed before running the `mkdir` command?

Reply:
We _could_, but `mkdir --parents` will do that for us.
It quietly succeeds if the dir exists.

Reply:
@flatcap 
But `mkdir` only in need is serveral times faster than running `mkdir -p`.
Just wrote a simple benchmark:
```bash
#!/usr/bin/env bash

for (( i = 0; i < 1000; i++ ))
do
    # ./mkdir_p.sh  1.16s user 0.72s system 103% cpu 1.823 total
    mkdir -p .git/info
done
```

```bash
#!/usr/bin/env bash

for (( i = 0; i < 1000; i++ ))
do
    # ./mkdir_if_need.sh  0.03s user 0.00s system 98% cpu 0.031 total
    if [ ! -d ".git/info" ]
    then
        mkdir .git/info
    fi
done
```

Maybe it is trivial to care about performance in shell script. But I write C and Lua in my daily job, so occupation disease, you know.

Reply:
> Maybe it is trivial to care about performance in shell script

hehe, not at all, but you have to pick your battles.
In this case, there's no loop so the time saving is too small to be noticed (by the human running it)
However, it **is** more efficient to `test` first, so I've updated the PR :-)